### ISSUE TITLE
10% kernel regression on normalization in upstream code bump

### ISSUE THREAD

**Issue by jjsjann123** [csarofeen/pytorch#1869]
### 🐛 Describe the bug

We noticed a 10% perf regression in microbenchmarks in last upstream push. https://github.com/pytorch/pytorch/pull/81861

I did a quick look at the kernel and I noticed that in the updated cuda kernel, we are getting more vectorized load. Kernel time went from ~26ms to ~29ms. (V100, measured with nsys for kernel time, which seems to be relatively stable.)

~Note that kernel time measuring is a little bit flaky using our bandwidth dump. This is somewhat surprising to me. With `PYTORCH_NVFUSER_DUMP=dump_eff_bandwidth`, the regression grows a long dramatic from 20ms to 40ms (A100).~ (nvm, can't repro this today any more. I was probably on a busted node.)

I have extracted the repro as vvv:

```
import time

import torch
import torch.utils.jit.log_extract as log_extract


model_str = """graph(%0 : Float(96, 160, 7, 7, strides=[7840, 49, 7, 1], requires_grad=0, device=cuda:0),
      %1 : Float(160, strides=[1], requires_grad=0, device=cuda:0),
      %2 : Float(160, strides=[1], requires_grad=0, device=cuda:0),
      %3 : Float(160, strides=[1], requires_grad=0, device=cuda:0),
      %4 : Float(160, strides=[1], requires_grad=0, device=cuda:0),
      %5 : Float(96, 160, 7, 7, strides=[7840, 49, 7, 1], requires_grad=0, device=cuda:0),
      %6 : Float(160, strides=[1], requires_grad=0, device=cuda:0),
      %7 : Float(160, strides=[1], requires_grad=0, device=cuda:0),
      %8 : Float(160, strides=[1], requires_grad=0, device=cuda:0),
      %9 : Float(160, strides=[1], requires_grad=0, device=cuda:0),
      %10 : Float(96, 160, 7, 7, strides=[7840, 49, 7, 1], requires_grad=0, device=cuda:0),
      %11 : Float(160, strides=[1], requires_grad=0, device=cuda:0),
      %12 : Float(160, strides=[1], requires_grad=0, device=cuda:0),
      %13 : Float(160, strides=[1], requires_grad=0, device=cuda:0),
      %14 : Float(160, strides=[1], requires_grad=0, device=cuda:0),
      %15 : int,
      %16 : int):
  %17 : float = prim::Constant[value=1.0000000000000001e-05]()
  %18 : float = prim::Constant[value=0.10000000000000001]()
  %19 : bool = prim::Constant[value=0]()
  %20 : Float(96, 160, 7, 7, strides=[7840, 49, 7, 1], requires_grad=0, device=cuda:0), %21 : Tensor, %22 : Tensor = aten::native_batch_norm(%10, %11, %12, %13, %14, %19, %18, %17)
  %23 : Float(96, 160, 7, 7, strides=[7840, 49, 7, 1], requires_grad=0, device=cuda:0), %24 : Tensor, %25 : Tensor = aten::native_batch_norm(%5, %6, %7, %8, %9, %19, %18, %17)
  %26 : Float(96, 160, 7, 7, strides=[7840, 49, 7, 1], requires_grad=0, device=cuda:0), %27 : Tensor, %28 : Tensor = aten::native_batch_norm(%0, %1, %2, %3, %4, %19, %18, %17)
  %29 : Float(96, 160, 7, 7, strides=[7840, 49, 7, 1], requires_grad=0, device=cuda:0) = aten::add(%26, %23, %16)
  %30 : Float(96, 160, 7, 7, strides=[7840, 49, 7, 1], requires_grad=0, device=cuda:0) = aten::add(%29, %20, %15)
  return (%30)
"""

_, inputs = log_extract.load_graph_and_inputs(model_str)
runtime = log_extract.run_nvfuser(model_str, inputs)
print("runtime with model: {}".format(runtime))
```

### Versions

Currently on the commit head in PR: https://github.com/pytorch/pytorch/pull/81861

So it's one/some of commits listed in that PR. Will give easier repro instruction once PR merged as a single commit.
